DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -24,56 +24,129 @@ public SectionContentService(IDbConnection connection, ILogger<SectionContentSer
+                            Applications.ApplicationName,
+                            Sections.SectionName, 
+                            Sections.DiagAssessPath,
+                            Sections.PLAssessPath,
+                            CustomisationTutorials.DiagStatus,
+                            Sections.SectionNumber,
+                            Sections.AverageSectionMins,
+                            Attempts.AttemptsPL,
+                            Attempts.PLPassed,
+                            Progress.ProgressId,
+                            Customisations.IsAssessed
+                        ) AS SectionContent
+
+                        INNER JOIN Tutorials ON SectionContent.SectionID = Tutorials.SectionID
+                        INNER JOIN CustomisationTutorials ON Tutorials.TutorialID = CustomisationTutorials.TutorialID
                         LEFT JOIN Progress
-                            ON Progress.CustomisationID = Customisations.CustomisationID
+                            ON Progress.CustomisationID = CustomisationTutorials.CustomisationID
                             AND Progress.CandidateID = @candidateId
                             AND Progress.RemovedDate IS NULL
                             AND Progress.SystemRefreshed = 0
                         LEFT JOIN aspProgress
                             ON aspProgress.TutorialID = CustomisationTutorials.TutorialID
-                            and aspProgress.ProgressID = Progress.ProgressID
-                    WHERE
-                        CustomisationTutorials.CustomisationID = @customisationId
-	                    AND Sections.SectionID = @sectionId
-	                    AND (Sections.ArchivedDate IS NULL)
-	                    AND (CustomisationTutorials.DiagStatus = 1 OR Customisations.IsAssessed = 1 OR CustomisationTutorials.Status = 1)
-                    GROUP BY
-                        Sections.SectionID,
-                        Customisations.CustomisationID,
-                        Progress.ProgressID,
-                        Progress.CustomisationID,
-                        Progress.CandidateID,
-                        Customisations.CustomisationName,
-                        Applications.ApplicationName,
-                        Sections.SectionName, 
-                        Sections.AverageSectionMins",
-                new { customisationId, candidateId, sectionId }
-            );
+                            AND aspProgress.ProgressID = Progress.ProgressID
+                        LEFT JOIN TutStatus
+                            ON aspProgress.TutStat = TutStatus.TutStatusID
+                        WHERE CustomisationTutorials.CustomisationID = @customisationId
+                        ORDER BY CustomisationTutorials.TutorialID",
+                    (section, tutorial) =>
+                {
+                    sectionContent ??= section;
+
+                    sectionContent.Tutorials.Add(tutorial);
```

Comment:
So the reason we need to do the nested select is because we want to sum various info from the tutorials right? If so, could we just do that manually here? As we're having to go manually go through each entry to add the tutorials anyway. And then we probably wouldn't need the nested select and could simplify this query? 🙂

Reply:
Would that work with Dapper Multimapping? I'm not sure exactly how it works but I assumed all of the section information would need to be the same for the fields

Reply:
So if we take the TutTime as an example, I think something like this would work:
* Set it to aspProgress.TutTime in the query
* Here when adding the tutorial to the section content, also add the TutTime. You'd have to be a bit careful not to add the first one twice, so something like:
```
if (sectionContent == null)
{
  sectionContent = section;
}
else 
{
  sectionContent.TutTime += section.TutTime;
}
```

What do you think? 🙂

Reply:
I think that works for most of them, there comes a problem when we go for Percent Complete as it isn't just a simple sum. For that would you recommend having a method within `SectionContent.cs` which takes the total percent complete (calculated in the same way as TutTime) and divides it by the total number of tutorials

Reply:
I'll also just have to check what all the other coalesces/sum/max statements do and then replicate them within the multimap

Reply:
For percentage complete I'd recommend changing SectionContent so that it calculates that based on the other fields (by using a custom getter). That can use the length of the tutorials list + hasLearning to calculate it. Does that make sense? 🙂